src/System.Device.Gpio/System/Device/I2c/II2cController.cs

```diff
@@ -0,0 +1,110 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+namespace System.Device.I2c
+{
+    public interface II2cController : IDisposable
+    {
+        /// <summary>
+        /// Determines if this should be disposed.
+        /// </summary>
+        bool ShouldDispose { get; set; }
+
+        /// <summary>
+        /// Opens an I2C device with specified settings in order for it to be ready to use with the controller.
+        /// </summary>
+        /// <param name="settings">The connection settings of a device on an I2C bus.</param>
+        /// <param name="shouldDispose">Determines if this should be disposed.</param>
+        void OpenDevice(I2cConnectionSettings settings, bool shouldDispose = true);
+
+        /// <summary>
+        /// Opens an I2C device in order for it to be ready to use with the controller.
+        /// </summary>
+        /// <param name="device">The I2C device to open.</param>
+        /// <param name="shouldDispose">Determines if this should be disposed.</param>
+        void OpenDevice(I2cDevice device, bool shouldDispose = true);
+
+        /// <summary>
+        /// Closes an open I2C device based on specified settings.
+        /// </summary>
+        /// <param name="settings">The connection settings of a device on an I2C bus.</param>
+        void CloseDevice(I2cConnectionSettings settings);
+
+        /// <summary>
+        /// Closes an open I2C device.
+        /// </summary>
+        /// <param name="device">The I2C device to close.</param>
+        void CloseDevice(I2cDevice device);
+
+        /// <summary>
+        /// Removes all I2C devices from controller.
+        /// </summary>
+        void ClearDevices();
+
+        /// <summary>
+        /// Gets the I2C device associated with the specified bus ID and address.
+        /// </summary>
+        /// <param name="busId">The bus ID the I2C device is connected to.</param>
+        /// <param name="address">The bus address of the I2C device to write.</param>
+        /// <param name="device">
+        /// When this method returns, contains the I2C device associated with the specified bus ID and address, if they are found;
+        /// otherwise, the default value for the type of the value parameter. This parameter is passed uninitialized.
+        /// </param>
+        bool TryGetDevice(int busId, int address, out I2cDevice? device);
+
```

Comment:
Should we also have a GetDevices() method that returns all opened devices?

Reply:
we could do that, and if so we should probably implement IEnumerable as well so that it may be used in a foreach. I'm fine with either that, or the GetDevices() method returning an IEnumerable<I2cDevice>

Reply:
Let's go with GetDevices().